device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/Message.java

```diff
@@ -370,5 +373,11 @@ public void setExpiryTime(long timeOut)
     {
         long currentTime = System.currentTimeMillis();
         this.expiryTime = currentTime + timeOut;
+        logger.LogInfo("The message with messageid %s has expiry time as %s, method name is %s ", this.getMessageId(), new Date(this.expiryTime), getMethodName());
+    }
+
+    private String getMethodName()
+    {
```

Comment:
Would it be possible to move getMethodName() to CustomLogger class aqnd avoid re-implementing it in every class ?